### PR TITLE
Container builds: Add a tag with commit sha

### PR DESCRIPTION
**Pull request by EricEdens**
This was motivated by https://github.com/GoogleCloudPlatform/guest-test-infra/pull/401 where I added SHA tags for Daisy; @hopkiw  suggested that I apply this to all containers.

### CODE REVIEW

File: concourse/pipelines/container-build.jsonnet [hunk 1]
```diff
@@ -8,6 +8,10 @@ local buildcontainerimgtask = {
   input:: error 'must set input in buildcontainerimgtask',
   context:: error 'must set context in buildcontainerimgtask',
   destination:: error 'must set destination in buildcontainerimgtask',
+  commit_sha:: error 'must set commit_sha in buildcontainerimgtask',
+
+  assert std.findSubstr(':', task.destination) == []
+         : 'Custom tags not supported; received: ' + task.destination,
```
Comment (hopkiw) [inline concourse/pipelines/container-build.jsonnet:14]: please remove as discussed

File: concourse/pipelines/container-build.jsonnet [hunk 2]
```diff
@@ -57,9 +62,14 @@ local buildcontainerimgjob = {
         ] +
         job.extra_steps +
         [
+          {
+            load_var: job.image + '-commit-sha',
+            file: job.input + '/.git/ref',
```
Comment (hopkiw) [inline concourse/pipelines/container-build.jsonnet:67]: for anything other than simple append, let's use formatting:
```suggestion
            file: '%s/.git/ref' % job.input,
```

**Review by hopkiw — CHANGES_REQUESTED** [review]

File: concourse/pipelines/container-build.jsonnet (hunk 2 above)
Comment (EricEdens) [reply concourse/pipelines/container-build.jsonnet]: done!

File: concourse/pipelines/container-build.jsonnet (hunk 1 above)
Comment (EricEdens) [reply concourse/pipelines/container-build.jsonnet]: removed

**Review by hopkiw — APPROVED** [review]

**Pull request merged**